src/types/account-manager.ts

```diff
@@ -257,22 +388,33 @@ export class AccountManager {
    * @returns The account
    */
   public random() {
-    return this.signerAccount(randomAccount())
+    return this.signerAccount(algosdk.generateAccount())
   }
 
   /**
-   * Returns an account (with private key loaded) that can act as a dispenser.
+   * Returns an account (with private key loaded) that can act as a dispenser from
+   * environment variables, or against default LocalNet if no environment variables present.
+   *
+   * Note: requires a Node.js environment to execute.
+   *
+   * It present, it will load the account mnemonic stored in process.env.DISPENSER_MNEMONIC and optionally
+   * process.env.DISPENSER_SENDER if it's a rekeyed account.
    *
    * @example
    * ```typescript
-   * const account = await account.dispenser()
+   * const account = await account.dispenserFromEnvironment()
    * ```
-   * If running on LocalNet then it will return the default dispenser account automatically,
-   *  otherwise it will load the account mnemonic stored in process.env.DISPENSER_MNEMONIC.
+   *
    * @returns The account
    */
-  public async dispenser() {
-    return this.signerAccount(await getDispenserAccount(this._clientManager.algod, this._clientManager.kmd))
+  public async dispenserFromEnvironment() {
+    if (!process || !process.env) {
+      throw new Error('Attempt to get dispenser from environment from a non Node.js context; this is not supported!')
+    }
+
+    return process.env[`${DISPENSER_ACCOUNT.toUpperCase()}_MNEMONIC`]
```

Comment:
This is technically a breaking change - previously we would get LocalNet if pointing to LocalNet, but I think it's much better to use environment variables in preference, not only is that less HTTP calls (no isLocalNet check), but it means that you can control the dispenser to be a different account if you want. I think the chances of this breaking someone in reality would be slim.